src/components/LocationListener.tsx

```diff
@@ -70,15 +70,20 @@ class LocationListener extends Component<Props> {
   }
   onLocation = (location: Location) => {
     const {
-      coords: {latitude, longitude},
+      coords: {latitude, longitude, speed},
       timestamp,
     } = location
+    const MPS_TO_MPH_RATE = 2.23694
+    const speedInMph = speed && speed * MPS_TO_MPH_RATE // meters/s -> mph
 
-    this.props.locationStorage.recordLocation({
-      latitude: latitude,
-      longitude: longitude,
-      when: timestamp,
-    })
+    // only recording locations user was moving under 10 mph at
+    if (speedInMph && speedInMph <= 10 || !!speed == false) { // !!speed coerces value of speed to a boolean, to account for null, undefined, 0, all which we want to record by default
```

Comment:
I guess i'm still hung up on `!!speed == false`. Aren't you alreading checking if speed exists earlier in the predicate where you say `speedInMph && ...`? Also, if it's definitely a boolean I think you'd want ` === false` or just `!speed`? I think we should pair on this to help alleviate my confusion.